wolfcrypt/test/test.c

```diff
@@ -18916,6 +18981,64 @@ int ed25519_test(void)
 #endif /* HAVE_ED25519_VERIFY */
     }
 
+    outlen = sizeof(out);
+    XMEMSET(out, 0, sizeof(out));
+    
+    if (wc_ed25519_import_private_key(sKeyCtx1, ED25519_KEY_SIZE, pKeyCtx1,
+                                      sizeof(pKeyCtx1), &key) != 0)
+        return -9020;
+
+    if (wc_ed25519ctx_sign_msg(msgCtx1, sizeof(msgCtx1), out, &outlen, &key,
+                               contextCtx1, sizeof(contextCtx1)) != 0)
+        return -9021;
+
+    if (XMEMCMP(out, sigCtx1, 64))
+        return -9022;
+
+#if defined(HAVE_ED25519_VERIFY)
+    /* test verify on good msg */
+    if (wc_ed25519ctx_verify_msg(out, outlen, msgCtx1, sizeof(msgCtx1), &verify,
+                                 &key, contextCtx1, sizeof(contextCtx1)) != 0 ||
+            verify != 1)
+        return -9023;
+
+    /* test verify on bad msg */
+    out[outlen-1] = out[outlen-1] + 1;
+    if (wc_ed25519ctx_verify_msg(out, outlen, msgCtx1, sizeof(msgCtx1), &verify,
+                                 &key, contextCtx1, sizeof(contextCtx1)) == 0 ||
+            verify == 1)
+        return -9024;
+#endif
+
+    outlen = sizeof(out);
+    XMEMSET(out, 0, sizeof(out));
+    
+    if (wc_ed25519_import_private_key(sKeyPh1, ED25519_KEY_SIZE, pKeyPh1,
+                                      sizeof(pKeyPh1), &key) != 0)
+        return -9025;
+
+    if (wc_ed25519ph_sign_msg(msgPh1, sizeof(msgPh1), out, &outlen, &key, NULL,
```

Comment:
Can we add a test for ph_sign_msg with a context?

Reply:
And both ph_sign_hash with and w/o a context?

Reply:
Added - no test vector in document
Added sign_hash and verify_hash tests.